python_modules/libraries/dagster-k8s/dagster_k8s_tests/unit_tests/test_ext_pod.py

```diff
@@ -0,0 +1,85 @@
+from dagster_k8s.external_resource import build_pod_body
+
+
+def test_pod_building():
+    # min info
+    build_pod_body(
+        pod_name="test",
+        image="test_image",
+        command=None,  # image provides default entrypoint
+        env_vars={},
+        base_pod_spec=None,
+        base_pod_meta=None,
+    )
+
+    # expected common case
+    build_pod_body(
+        pod_name="test",
+        image="test_image",
+        command=["echo", "hello"],
+        env_vars={},
+        base_pod_spec=None,
+        base_pod_meta=None,
+    )
+
+    # spec provided
+    build_pod_body(
+        pod_name="test",
+        image=None,
+        command=None,
+        env_vars={},
+        base_pod_spec={
+            "containers": [
+                {
+                    "image": "test_image",
+                }
+            ]
+        },
+        base_pod_meta=None,
+    )
```

Comment:
seems like we should assert something about the outputs here since we are varying inputs